### PR TITLE
Colored Output and Pretty Formatting

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:* https://github.com/aws-cloudformation/cfn-python-lint/issues/971

*Description of changes:*
- Add pretty formatter that groups outputs by file name and adds color for terminals that support ansi

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by PatMyron** [comment]
Wondering about sorting violations by severity instead of just line number:

Screen Shot 2020-10-21 at 1 13 50 PM

File: src/cfnlint/formatters/__init__.py [hunk 1]
```diff
@@ -2,11 +2,29 @@
...
 
 
+class color(object):
+    error = '\033[31m'
+    warning = '\033[33m'
+    informational = '\033[34m'
+    unknown = '\033[37m'
+    green = '\033[32m'
+    reset = '\033[0m'
+    bold_green = '\033[1;32m'
```
Comment (PatMyron) [inline src/cfnlint/formatters/__init__.py:20]: Couldn't find much shell color documentation more authoritative than https://gist.github.com/vratiu/9780109

Any other resource you used?

Comment (kddejong) [reply src/cfnlint/formatters/__init__.py:20]: Not really.  https://github.com/tartley/colorama/blob/master/colorama/ansi.py#L49-L58 looked at other color packages too.

**Comment by kddejong** [comment]
Interesting.  Absolutely we could sort by severity.

**Comment by PatMyron** [comment]
Also wondering if [summaries](https://github.com/aws-cloudformation/cfn-python-lint/issues/882) should be dealt with in the same format as [color](https://github.com/aws-cloudformation/cfn-python-lint/issues/971)

**Comment by kddejong** [comment]
Good call.  We could add that to this one.

**Comment by kddejong** [comment]
I got it down to something like this.
`Cfn-lint scanned 11 templates against 127 rules and found 7 errors, 3 warnings, and 0 informational violations`

thoughts?

**Comment by kddejong** [comment]
Current output view.

image

**Comment by kddejong** [comment]
not sure the line and column numbers need to be green either.

**Comment by kddejong** [comment]
Line and column numbers are now white

**Comment by PatMyron** [comment]
Not a big deal, but heads up that tab spacing can lead to cliffs for different number of digits in line numbers:

Screen Shot 2020-10-28 at 12 47 50 PM

**Comment by kddejong** [comment]
@PatMyron fixed.

File: src/cfnlint/formatters/__init__.py [hunk 2]
```diff
@@ -172,3 +188,60 @@ def _format(self, match):
...
+
+class PrettyFormatter(BaseFormatter):
+    """Generic Formatter"""
+
+    def _format(self, match):
+        """Format output"""
+        formatstr = '{0}{1}{2}'
+        pos = '{0}:{1}:'.format(match.linenumber, match.columnnumber)
+        return formatstr.format(
+            colored('{:20}'.format(pos), color.reset),
```
Comment (kddejong) [inline src/cfnlint/formatters/__init__.py:201]: `{:20}` will keep the width without tabs

**Pull request merged**